joomla/joomla-cms
# [4.0] Restores the possibility to show rule descriptions in rules formfield.

With https://github.com/joomla/joomla-cms/pull/19285, the possibility to add a description to a custom rule was removed.
@wilsonge reverted part of the change so it should still be possible to set a description, ~but he made a typo in his revert~ :smile: 

### Summary of Changes
This PR fully restores the possibility to set and display a description in the rules field.
It incorporates a check for the presence of the description so the rules that no longer have a description don't show a tooltip.

### Testing Instructions
* Make sure the rules in core extensions don't show any description. They should all be removed with the original PR and still don't show any description.
* Test with a 3rd party extension (eg https://github.com/Bakual/SermonSpeaker/releases/download/6.0.0-alpha1/com_sermonspeaker.zip) that rules with a description still show a tooltip OR add a description to the rules in access.xml file of com_content and test if it shows up.

### Expected result
A description is shown
Please note that the tooltip itself is broken due to Bootstrap tooltips not being loaded on that page. The method used in core is `JHtml::tooltipText()` which doesn't load Bootstrap by itself. Same issue exists in other places (eg extension manager) as well and needs to be fixed independant of this PR.
The description will show as a native "title".

### Actual result
Nothing is shown


### Documentation Changes Required
None
